src/main/java/com/atlassian/jira/cloud/jenkins/deploymentinfo/pipeline/JiraSendDeploymentInfoStep.java

```diff
@@ -0,0 +1,177 @@
+        this.site = site;
+    }
+
+    public String getEnvironmentId() {
+        return environmentId;
+    }
+
+    @DataBoundSetter
+    public void setEnvironmentId(final String environmentId) {
+        this.environmentId = environmentId;
+    }
+
+    public String getEnvironmentName() {
+        return environmentName;
+    }
+
+    @DataBoundSetter
+    public void setEnvironmentName(final String environmentName) {
+        this.environmentName = environmentName;
+    }
+
+    public String getEnvironmentType() {
+        return environmentType;
+    }
+
+    @DataBoundSetter
+    public void setEnvironmentType(final String environmentType) {
+        this.environmentType = environmentType;
+    }
+
+    @Override
+    public StepExecution start(final StepContext stepContext) throws Exception {
+        return new JiraSendDeploymentInfoStepExecution(stepContext, this);
+    }
+
+    @Extension
+    public static class DescriptorImpl extends StepDescriptor {
+
+        @Inject private transient JiraCloudPluginConfig globalConfig;
+
+        @Override
+        public Set<Class<?>> getRequiredContext() {
+            return ImmutableSet.of(TaskListener.class, Run.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "jiraSendDeploymentInfo";
+        }
+
+        @Override
+        public String getDisplayName() {
+            return Messages.JiraSendDeploymentInfoStep_DescriptorImpl_DisplayName();
+        }
+
+        @SuppressWarnings("unused")
+        public ListBoxModel doFillSiteItems() {
+            ListBoxModel items = new ListBoxModel();
+            final List<JiraCloudSiteConfig> siteList = globalConfig.getSites();
+            for (JiraCloudSiteConfig siteConfig : siteList) {
+                items.add(siteConfig.getSite(), siteConfig.getSite());
+            }
+
+            return items;
+        }
+
+        @SuppressWarnings("unused")
+        public ListBoxModel doFillEnvironmentTypeItems() {
+            ListBoxModel items = new ListBoxModel();
+            items.add("development", "development");
+            items.add("testing", "testing");
+            items.add("staging", "staging");
+            items.add("production", "production");
+
+            return items;
+        }
+    }
+
+    public static class JiraSendDeploymentInfoStepExecution
+            extends SynchronousNonBlockingStepExecution<JiraSendInfoResponse> {
+
+        private final JiraSendDeploymentInfoStep step;
+
+        public JiraSendDeploymentInfoStepExecution(
+                final StepContext context, final JiraSendDeploymentInfoStep step) {
+            super(context);
+            this.step = step;
+        }
+
+        @Override
+        protected JiraSendInfoResponse run() throws Exception {
+            final TaskListener taskListener = getContext().get(TaskListener.class);
+            final WorkflowRun workflowRun = (WorkflowRun) getContext().get(Run.class);
```

Comment:
/nit you can get the WorkflowRun from the context without a cast `getContext().get(WorkflowRun.class)` I don't remember why I was using the `Run` class in the BuildInfoSender instead of using `WorkflowRun`

Reply:
I have updated this for Builds as well.